src/components/DataGrid/utils.tsx

```diff
@@ -104,81 +119,52 @@ export const createSingleGridColDef: CreateSingleGridColDefFunction = (columnLab
-      width: 40,
-      hideable: false,
-      getActions: (params) => [
-        <Tooltip title={`Open ${params.row.omschrijving}`}>
-          <GridActionsCellItem
-            color="primary"
-            icon={<ArrowForward />}
-            label={`Open ${params.row.omschrijving}`}
-            onClick={handleClick(params)}
-          />
-        </Tooltip>,
-      ],
-    },
-    {
-      field: 'omschrijving',
-      headerName: 'Omschrijving',
-      description: 'omschrijving',
-      type: 'string',
-      width: 220,
-      minWidth: 220,
-      flex: 2,
-      hideable: !defaultColumns.includes('omschrijving'),
-      renderHeader: () =>
-        loading ? (
-          <Skeleton variant="text" width={80} height={26} />
-        ) : (
-          <GridColumnHeaderTitle
-            label="Omschrijving"
-            columnWidth={220}
-            description="Omschrijving"
-          />
-        ),
-    },
-    {
-      field: 'identificatie',
-      headerName: 'Identificatie',
-      description: 'identificatie',
-      type: 'string',
-      minWidth: 220,
-      flex: 1,
-      hideable: !defaultColumns.includes('identificatie'),
-      renderHeader: () =>
-        loading ? (
-          <Skeleton variant="text" width={80} height={26} />
-        ) : (
-          <GridColumnHeaderTitle
-            label="Identificatie"
-            columnWidth={220}
-            description="Identificatie"
-          />
-        ),
-    },
-    ...(Object.entries(resultWithTypes)
-      .map(([columnLabel, columnType]) =>
-        createSingleGridColDef(columnLabel as keyof ZaaktypeT | keyof InformatieObjectT, columnType)
-      )
-      .filter((x) => x !== undefined) as GridColDef[]),
+export function getInitialData(
+  zaaktype?: ZaaktypeResolvedT,
+  informatieobjecttypen?: InformatieObjectT[]
+) {
+  const infoObjectTypes = zaaktype?.informatieobjecttypen;
+  const allInfoObjectTypes = informatieobjecttypen;
+
+  // If there is no zaaktype, return empty data
+  if (!infoObjectTypes) {
+    return { rows: [], selection: [], zaaktype: zaaktype };
+  }
+
+  // if there is no informatieobjecttypen, return zaaktype data
+  if (!allInfoObjectTypes) {
+    const infoObjectTransformed = infoObjectTypes.map((row: any) => createRow(row));
+    return { rows: infoObjectTransformed, selection: [], zaaktype: zaaktype };
+  }
+
+  // Merge zaaktype data with informatieobjecttypen data and exclude duplicates.
+  const existingRelations = infoObjectTypes.map((row: any) => createRow(row));
+  const allData = allInfoObjectTypes.map((row) => createRow(row));
+
+  const selectedRows = existingRelations.map((item: any) => item.id);
+
+  const mergedData = [
+    ...existingRelations,
+    ...allData.filter((row) => !selectedRows.includes(row.id)),
   ];
-};
+
+  return { rows: mergedData, selection: selectedRows, zaaktype: zaaktype };
+}
+
+// Create row and add extra fields to prevent errors
+export function createRow(row: any) {
+  const flattedRow = flat(row);
```

Comment:
Yes it is needed to be able to display the `volgnummer`, `richting` and `statustype`. Because the grid can't display properties nested inside an object. The flat flatten the object to make sure there are no nested objects inside.

Reply:
Ahh I see